src/conn.rs

```diff
@@ -440,6 +446,12 @@ impl<'a, 'c> InProgress<'a, 'c> {
         metadata.generation += 1;
         metadata.partition_map = self.partition_map;
 
+        // let the transaction observer know that there have been some transactions committed.
```

Comment:
Do this after the cache update and the schema switchover. Indeed, merge the two new `if let` blocks!